geoips/plugin_registry.py

```diff
@@ -19,6 +19,121 @@
 LOG = logging.getLogger(__name__)
 
 
+class PluginRegistry:
+    """Plugin Registry class definition.
+
+    Represents all of the plugins found in all of the available GeoIPS packages.
+    This class will load a plugin when requested, rather than loading all plugins when
+    GeoIPS is instantiated.
+    """
+    def __init__(self, _test_registry_files=[]):
+        # Use this for unit testing
+        if _test_registry_files:
+            self.registry_files = _test_registry_files
+        # Use this for normal operation and collect the registry files
+        else:
+            self.registry_files = []  # Collect the paths to the registry files here
+            for pkg in get_entry_point_group("geoips.plugin_packages"):
+                self.registry_files.append(
+                    str(resources.files(pkg.value) / "registered_plugins.yaml")
+                )
+
+    @property
+    def registered_plugins(self):
+        """Find all plugins in registered plugin packages.
+
+        Search the ``registered_plugins.yaml`` of each registered plugin package and add
+        them to the _registered_plugins dictionary
+        """
+        # Load the registries here and return them as a dictionary
+        if not hasattr(self, "_registered_plugins"):
+            self._registered_plugins = {}
+            self._interface_mapping = {}
```

Comment:
Since these attributes are important, it would  be nice to have comments describing what they contain.

Reply:
I can update those with comments.